src/HttpHeader.cc

```diff
@@ -759,6 +774,16 @@ HttpHeader::addEntry(HttpHeaderEntry * e)
 
     debugs(55, 7, this << " adding entry: " << e->id << " at " << entries.size());
 
+    // verify whether adding the entry will not exceed the maximum header size allowed
+    if (const auto max = maxLen()) {
+        const auto delta = e->length();
```

Comment:
Nit: `delta` makes me think it might go negative (+/-).

```suggestion
        const auto eLen = e->length();
```